webapi/src/main/scala/org/knora/webapi/store/cache/impl/CacheServiceLive.scala

```diff
@@ -35,10 +35,10 @@ import org.knora.webapi.store.cache.api.EmptyValue
  * @param projects a map of projects.
  * @param lut   a lookup table of username/email to IRI.
  */
-case class CacheServiceInMemImpl(
+case class CacheServiceLive(
   users: TMap[String, User],
   projects: TMap[String, ProjectADM],
-  lut: TMap[String, String] // sealed trait for key type
+  lut: TMap[String, String]
```

Comment:
I know you're not touching the naming now, but I do want to note that `lut` might be improved upon by `mappingToEmail`.

Reply:
Thanks for pointing this out. This actually made me think and I have split up the `lut` into different mappings.

The `lut` (I guess it means `lookUpTable`) contained all kinds of mappings from `email`, `username`, `shortcode` to IRI of `User` or `ProjectADM`.

For example in case there is a `User.Username` which equals a `ProjectADM.Shortname`this caching was broken. I have fixed that in commit b0360fdc518719845b7610cd4dca2492419ee4b5

Reply:
Cool. I suppose another way would've been to embrace the untypedness, but adding a ClassTag, storing it alongside and making it truly generic. Too bad it can't be restricted to a union type yet.